compliant/src/lib.rs

```diff
@@ -0,0 +1,1171 @@
+                    println!("{}{}", prefix, cuda_variant.dimmed());
+                }
+            }
+
+            // Only show ROCm section if the feature is enabled
+            #[cfg(feature = "enable_rocm")]
+            {
+                println!("│  {} {}", rocm_mark, "ROCM".bold());
+
+                for (i, rocm_variant) in rocm_variants.iter().enumerate() {
+                    let is_last = i == rocm_variants.len() - 1;
+                    let is_present = rocm_variants_present.contains(&rocm_variant);
+                    let prefix = if is_last {
+                        "│    ╰── "
+                    } else {
+                        "│    ├── "
+                    };
+
+                    if is_present {
+                        println!("{}{}", prefix, rocm_variant);
+                    } else {
+                        println!("{}{}", prefix, rocm_variant.dimmed());
+                    }
+                }
+            }
+        } else {
+            // Compact output
+            #[cfg(feature = "enable_rocm")]
+            {
+                println!("│   ├── {} CUDA", cuda_mark);
+                println!("│   ╰── {} ROCM", rocm_mark);
+            }
+
+            #[cfg(not(feature = "enable_rocm"))]
+            {
+                println!("│   ╰── {} CUDA", cuda_mark);
+            }
+        }
+
+        // ABI status section
+        println!("╰── abi: {}", abi_status);
+        println!("    ├── {} {}", abi_mark, abi_output.manylinux_version);
+        println!(
+            "    ╰── {} python {}",
+            abi_mark, abi_output.python_abi_version
+        );
+    }
+}
+
+#[derive(Serialize)]
+pub struct RepoErrorResponse {
+    repository: String,
+    status: String,
+    error: String,
+}
+
+#[derive(Serialize)]
+pub struct RepositoryCheckResult {
+    repository: String,
+    status: String,
+    build_status: BuildStatus,
+    abi_status: AbiStatus,
+}
+
+#[derive(Serialize)]
+pub struct BuildStatus {
+    summary: String,
+    cuda: CudaStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    rocm: Option<RocmStatus>,
+}
+
+#[derive(Serialize)]
+pub struct CudaStatus {
+    compatible: bool,
+    present: Vec<String>,
+    missing: Vec<String>,
+}
+
+#[derive(Serialize)]
+pub struct RocmStatus {
+    compatible: bool,
+    present: Vec<String>,
+    missing: Vec<String>,
+}
+
+#[derive(Serialize)]
+pub struct AbiStatus {
+    compatible: bool,
+    manylinux_version: String,
+    python_abi_version: String,
+    variants: Vec<VariantCheckOutput>,
+}
+
+#[derive(Serialize)]
+pub struct VariantCheckOutput {
+    name: String,
+    compatible: bool,
+    has_shared_objects: bool,
+    violations: Vec<String>,
+}
+
+pub fn get_cache_dir() -> Result<PathBuf> {
+    let cache_dir = if let Ok(dir) = std::env::var("HF_KERNELS_CACHE") {
+        PathBuf::from(dir)
+    } else {
+        dirs::home_dir()
+            .unwrap_or_else(std::env::temp_dir)
+            .join(".cache/huggingface/hub")
```

Comment:
We should avoid custom logic. The user could have set `HF_HOME` and this would not be a good cache directory on Windows. If no specific kernels cache directory is set, we should revert to the default behavior of hf-hub.

Reply:
agreed, updated to prefer using hf-hub to handle the cache location